MM2RandoLib/Randomizers/RWeaknesses.cs

```diff
@@ -381,29 +535,31 @@ private void RandomizeWilyUJ(Patch in_Patch, ISeed in_Seed)
                     busterDmg = (Byte)(in_Seed.NextInt32(5) + 3);
                 }
                 in_Patch.Add(EDmgVsEnemy.DamageP + EDmgVsEnemy.Offset.PicopicoKun, busterDmg, String.Format("Buster Damage to Picopico-Kun"));
-                WilyWeaknesses[1, 0] = busterDmg;
+                WilyWeaknesses[EBossIndex.Pico][EWeaponIndex.Buster] = busterDmg;
 
                 // Deal ammoUse x 10 for the main weakness
                 // Deal ammoUse x 6 for another
                 // Deal ammoUse x 3 for another
-                List<EDmgVsEnemy> pico = new List<EDmgVsEnemy>(dmgPtrEnemies);
-                rInt = in_Seed.NextInt32(pico.Count);
-                enemyWeak1 = pico[rInt];
-                pico.RemoveAt(rInt);
-                rInt = in_Seed.NextInt32(pico.Count);
-                enemyWeak2 = pico[rInt];
-                pico.RemoveAt(rInt);
-                rInt = in_Seed.NextInt32(pico.Count);
-                enemyWeak3 = pico[rInt];
-                for (Int32 i = 0; i < dmgPtrEnemies.Count; i++)
+                Dictionary<EWeaponIndex, EDmgVsEnemy> pico = new(dmgPtrEnemies);
+
+                //1st weakness
+                enemyWeak1 = in_Seed.NextElementAndRemove(pico);
+
+                //2nd weakness
+                enemyWeak2 = in_Seed.NextElementAndRemove(pico);
+
+                //3rd weakness
+                enemyWeak3 = in_Seed.NextElementAndRemove(pico);
+
+                foreach (EWeaponIndex i in dmgPtrEnemies.Keys)
```

Comment:
With the `weak` variables now being KeyValuePairs, some of the loops below can be simplified but I'll leave that as an exercise for the reader.